### PR TITLE
better fix for Maximum call stack size exceeded

### PR DESCRIPTION
**Pull request by enjalot**
I've implemented a more thorough handling for tooltips going outside of the window. The problem occurs in both float or solid mode when updating the place of the tooltip will also put it outside the window, this can happen with items (like icons) placed in a corner of the screen.

This code will only update the place of the tooltip if updating it will not put the tooltip outside in anyway. In most cases this should still fix the positioning, but for the rare cases that it doesnt it will simply allow the tooltip to go outside of the window rather than erroring with Maximum call stack size exceeded.

I've re-enabled the functionality for solid since this should always work.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -394,32 +394,67 @@ export default class ReactTooltip extends Component {
...
+    function getStyleLeft (_place) {
+      var _offsetEffectX = effect === 'solid' ? 0 : _place ? offsetFromEffect[_place].x : 0
+      return x + _offsetEffectX + xPosition
+    }
+    function getStyleTop (_place) {
+      var _offsetEffectY = effect === 'solid' ? 0 : _place ? offsetFromEffect[_place].y : 0
+      return y + _offsetEffectY + yPosition
+    }
+
+    var windoWidth = window.innerWidth
```
Comment (pavelloz) [inline src/index.js]: @enjalot Little typo :)

Comment (enjalot) [inline src/index.js]: hah i didn't notice. nice way to save a character ;)

**Comment by wwayne** [comment]
Thanks :+1: 
Gonna make some small change(code style, es6...) before update npm

**Comment by wwayne** [comment]
@enjalot  NPM published, v1.0.0

**Comment by enjalot** [comment]
awesome! :100: thanks so much.

**Pull request merged**